lib/elixir/lib/map.ex

```diff
@@ -425,7 +441,12 @@ defmodule Map do
   """
   @spec drop(map, [key]) :: map
   def drop(map, keys) do
-    Enum.reduce(keys, map, &delete(&2, &1))
+    do_drop(keys, map)
```

Comment:
I assume this `@spec` is wrong as well, we do need to handle any enumerable, right?
Same with `split/2`.